### PR TITLE
[ENH] CredentialManager: Store passwords in System Keyring Services

### PR DESCRIPTION
**Pull request by nikicc**
Credential Manager is used for safely storing passwords in system's keyring services.

### CODE REVIEW

**Comment by nikicc** [comment]
Short example added.

File: Orange/widgets/credentials.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @property
+    def key(self):
+        return keyring.get_password(self.service_name, self.username)
+
+    @key.setter
+    def key(self, value):
+        keyring.set_password(self.service_name, self.username, value)
+
+    def delete_password(self):
+        keyring.delete_password(self.service_name, self.username)
```
Comment (kernc) [inline Orange/widgets/credentials.py]: Can these methods be replaced with `__setattr__`, `__getattr__`, and `__delattr__` so the supported interface becomes more like a general secure key-value store?

File: Orange/widgets/credentials.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import keyring
+
+SERVICE_NAME = 'Orange3 - {}'
+
+
+class CredentialManager:
+    def __init__(self, username):
+        """
```
Comment (kernc) [inline Orange/widgets/credentials.py]: Class docstring goes outside `__init__`.

File: Orange/widgets/credentials.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        Class for storing passwords in the system keyring service.
+
+        Args:
+            username: username used for storing a matching password.
+
+        Examples:
+        >>> cm = CredentialManager('Foo')
```
Comment (kernc) [inline Orange/widgets/credentials.py]:
```py
cm = CredentialManager('Some Widget')
```

File: Orange/widgets/credentials.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        >>> cm.key = 'Bar'      # store password
+        >>> cm.key              # retrieve password
+        'Bar'
+        """
+        self.username = username
+        self.service_name = SERVICE_NAME.format(self.username)
+
+    @property
+    def key(self):
+        return keyring.get_password(self.service_name, self.username)
```
Comment (kernc) [inline Orange/widgets/credentials.py]: Should getter return None by default?

File: Orange/widgets/credentials.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import keyring
+
+SERVICE_NAME = 'Orange3 - {}'
+
+
+class CredentialManager:
+    def __init__(self, username):
```
Comment (kernc) [inline Orange/widgets/credentials.py]: Service name (namespace), not username (key).

**Comment by nikicc** [comment]
@kernc please check again.

**Pull request merged**